server/character.lua

```diff
@@ -69,13 +69,18 @@ lib.callback.register('qbx_core:server:createCharacter', function(source, data)
 
     giveStarterItems(source)
     if GetResourceState('qbx_spawn') == 'missing' then
-        SetPlayerRoutingBucket(source, 0)
+        SetPlayerBucket(source, 0)
     end
 
     lib.print.info(('%s has created a character'):format(GetPlayerName(source)))
     return newData
 end)
 
+lib.callback.register('qbx_core:server:characterBucket', function (source)
+    SetPlayerBucket(source, source)
+    return GetPlayerRoutingBucket(source) == source
```

Comment:
Under what scenario would setting the bucket fail?

Reply:
Would it be better as an event rather than a callback in that case?

Reply:
I think the callback is fine. Callbacks don't have to return anything, but you are doing logic in response to the return. If this is just a safety check, then assert it here rather than return.

Reply:
@Manason i used callback.await, so isn't the return necessary?

Reply:
I believe if there is no return specified, a default return of nil will be present.

Reply:
I can revert before latest commits and go back to .await then, just return true regardless and assert error for the bucket check.